packages/material-ui/src/Theme/MaterialUIContextProps.ts

```diff
@@ -1,69 +1,60 @@
-/** Use require for loading these libraries in case they are not available in order to perform a useful fallback
- */
-let mui = {};
-try {
-  mui = require('@material-ui/core');
-} catch (err) {
-  // purposely a no-op
-}
-
-// @ts-ignore What we are doing here isn't really good Typescript, but it works
-const {
-  Box,
-  Button,
-  Checkbox,
-  Divider,
-  FormControl,
-  FormControlLabel,
-  FormGroup,
-  FormHelperText,
-  FormLabel,
-  Grid,
-  IconButton,
-  Input,
-  InputLabel,
-  List,
-  ListItem,
-  ListItemIcon,
-  ListItemText,
-  MenuItem,
-  Paper,
-  Radio,
-  RadioGroup,
-  Slider,
-  SvgIcon,
-  TextField,
-  Typography,
-} = mui;
+import { ComponentType } from 'react';
+import type {
+  BoxProps,
+  ButtonProps,
+  CheckboxProps,
+  DividerProps,
+  GridProps,
+  FormControlProps,
+  FormControlLabelProps,
+  FormGroupProps,
+  FormHelperTextProps,
+  FormLabelProps,
+  IconButtonProps,
+  InputProps,
+  InputLabelProps,
+  ListProps,
+  ListItemProps,
+  ListItemIconProps,
+  ListItemTextProps,
+  MenuItemProps,
+  PaperProps,
+  RadioProps,
+  RadioGroupProps,
+  SliderProps,
+  SvgIconProps,
+  TextFieldProps,
+  TypographyProps,
+} from '@material-ui/core';
 
 export default interface MaterialUIContextProps {
-  Box?: Box;
-  Button?: Button;
-  Checkbox?: Checkbox;
-  Divider?: Divider;
-  Grid?: Grid;
-  FormControl?: FormControl;
-  FormControlLabel?: FormControlLabel;
-  FormGroup?: FormGroup;
-  FormHelperText?: FormHelperText;
-  FormLabel?: FormLabel;
-  IconButton?: IconButton;
-  Input?: Input;
-  InputLabel?: InputLabel;
-  List?: List;
-  ListItem?: ListItem;
-  ListItemIcon?: ListItemIcon;
-  ListItemText?: ListItemText;
-  MenuItem?: MenuItem;
-  Paper?: Paper;
-  Radio?: Radio;
-  RadioGroup?: RadioGroup;
-  Slider?: Slider;
-  TextField?: TextField;
-  Typography?: Typography;
-  AddIcon?: SvgIcon;
-  ArrowDownwardIcon?: SvgIcon;
-  ArrowUpwardIcon?: SvgIcon;
-  ErrorIcon?: SvgIcon;
-  RemoveIcon?: SvgIcon;
+  Box: ComponentType<BoxProps>;
```

Comment:
I noticed that typescript was giving me an error for each component when using this approach. Complaining about not having a constructor... It still seems to work, but that's a lot of TS error to potentially clean up if we ever enable full typescript error validation.

Reply:
Interesting. I'm not seeing those errors/warnings in VS code or during build.

Reply:
I'll pull down your PR and see if I still get them